src/ActionEventEmitterEventStore.php

```diff
@@ -221,11 +221,11 @@ public function appendTo(StreamName $streamName, Iterator $streamEvents): void
         $this->actionEventEmitter->dispatch($event);
 
         if ($event->getParam('streamNotFound', false)) {
-            throw StreamNotFound::with($streamName);
+            throw $event->getParam('streamNotFound');
         }
 
         if ($event->getParam('concurrencyException', false)) {
-            throw new ConcurrencyException();
+            throw $event->getParam('concurrencyException');
```

Comment:
why get twice? just do `if ($exception = $event->getParam('concurrencyException', false)) {`

Reply:
I don't really like assignments in ifs but sure, I'll change it.